src/inverseRobot.js

```diff
@@ -3,11 +3,25 @@
 /*
  * @param {object} robot
  *
- * @return {object}
+ * @return {object | null}
 */
 
 function inverseRobot(robot) {
-  // write code here
+  const fixed = {};
+  const valuesSet = new Set();
+
+  for (const key in robot) {
+    const value = robot[key];
+
+    if (valuesSet.has(value)) {
```

Comment:
I really appreciate that you are going forward and that you are using such things as Set.
But I think that in this case it's not the way. You are storing all properties in two places: fixed and valuesSet.
You could do it only using regular object and as shown in the checklist - use hasOwnProperty.
But if you really want to use some more advanced JS features, I would get rid of fixedRobot object and use Map. At the end of the function convert the map to an object and return it.

Reply:
So if you want more complex solution - google Map and converting it to an object ;)